### PR TITLE
DCGEO-65 (DRAFT): pre adj data feature flag

### PR DESCRIPTION
**Pull request by KyRobbins**
### Change Details

Adds a feature flag to allow toggling the serving of Pre Adjudicated claim resources via the fhir API using environment variables passed in as system properties.

### Acceptance Validation

Is the implementation of this feature flag consistent with implementations of similar features?

### Feedback Requested

Are there any areas where this implementation does not fully cover the needs as would be required by such a feature flag?

### External References

- [DCGEO-65](https://jira.cms.gov/browse/DCGEO-65)

### Security Implications

No extra PII/PHI concerns.

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

File: ops/ansible/playbooks-ccs/vars/test/group_vars/all/env_specific.yml [hunk 1]
```diff
@@ -20,6 +20,9 @@ data_pipeline_idempotency_required: false
 # Whether to enable BFD API V2
 data_server_v2_enabled: true
 
+# Whether to enable BFD PreAdj Resources
+pre_adj_resources_enabled: false
```
Comment (ericbuckley) [inline ops/ansible/playbooks-ccs/vars/test/group_vars/all/env_specific.yml:24]: @KyRobbins depending on the order in which the PRs get merged (this, search and transformers), we can set this to `true`.

**Review by ericbuckley — DISMISSED** [review]
The changes specific to this PR look good, but lets wait for #612 to merge before opening this up for review

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/SpringConfiguration.java [hunk 2]
```diff
@@ -340,6 +343,19 @@ public PersistenceAnnotationBeanPostProcessor persistenceAnnotationProcessor() {
...
+   * Determines if the fhir resources related to pre adj claim data should be accessible via the
+   * fhir api service.
+   *
+   * @return True if the resources should be available to consume, False otherwise.
+   */
+  private static boolean isPreAdjResourcesEnabled() {
+    return Boolean.TRUE
+        .toString()
+        .equalsIgnoreCase(
+            ObjectUtils.defaultIfNull(System.getProperty("bfdServer.preadj.enabled"), "false"));
```
Comment (brianburton) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/SpringConfiguration.java:356]: Meaning would be clearer using `System.getProperty("bfdServer.preadj.enabled", "false")`

**Review by brianburton — APPROVED** [review]
Looks good to me.

**Comment by Gabasol** [comment]
Per @KyRobbins's slack message, this PR has been replaced by https://github.com/CMSgov/beneficiary-fhir-data/pull/667.

**Review by brianburton — APPROVED** [review]

**Comment by KyRobbins** [comment]
Replaced by https://github.com/CMSgov/beneficiary-fhir-data/pull/667

**Pull request closed without merging**